src/lib/vocab/entity/entityHelpers.server.ts

```diff
@@ -47,3 +48,63 @@ export const updateDirectories = async (
 		directoryIds.map(async (directoryId) => repos.entity.update(directoryId) as Promise<Directory>),
 	);
 };
+
+/** This helper function takes a newly created entity & tie and checks
+ * to see if the source entity needs its orderedItems attribute updated
+ *
+ * @param repos - the db repos
+ * @param createdEntity - the newly created entity, may or may not have orderedItems
+ * @param tie - the newly created relationship
+ * @returns either the updated entity with orderedItems or void if no change
+ */
+export const checkAddOrderedItem = async (
+	repos: Repos,
+	createdEntity: Entity,
+	tie: Tie,
+): Promise<Entity | void> => {
+	if (tie.type !== 'HasItem') return;
+	const source =
+		createdEntity.entity_id === tie.source_id
+			? createdEntity
+			: await repos.entity.findById(tie.source_id);
+	if (source && 'orderedItems' in source.data) {
+		if (!source.data.orderedItems) {
+			source.data.orderedItems = [];
+		}
```

Comment:
`orderedItems` isn't nullable so I think this is right:

```suggestion
	if (source?.data.orderedItems) {
```